### PR TITLE
Fix dependency_graph-plugin

### PR DESCRIPTION
**Pull request by wuarmin**
Hey,

i stumbled upon an issue while using the dependency_graph-plugin:
```txt
undefined method `to_sym' for #<Dry::System::Identifier:0x000056140520e7b0>
# ./lib/dry/system/plugins/dependency_graph.rb:38:in `register'
```
This changes would fix my issue. In addition, the event keys are now all symbols. That wasn't the case. The dependency_graph-spec has also been improved.

What do you think?

kind regards

### CODE REVIEW

**Review by solnic — CHANGES_REQUESTED** [review]
Thanks for tackling this. Unfortunately what should happen is ensuring those are always strings. Working with symbols like ":'foo.bar.baz'" is not nice 🙂

File: lib/dry/system/plugins/dependency_graph.rb [hunk 1]
```diff
@@ -34,10 +34,10 @@ def self.dependencies
         # @api private
         def register(key, contents = nil, options = {}, &block)
           super
-
-          unless config.ignored_dependencies.include?(key.to_sym)
```
Comment (wuarmin) [inline lib/dry/system/plugins/dependency_graph.rb:38]: Alright, so `ignored_dependencies` should also be an array of strings?

Comment (solnic) [reply lib/dry/system/plugins/dependency_graph.rb:38]: Yes definitely! dry-system allows strings as identifiers in auto-injection modules BUT they are all coerced to strings internally, and that's how **dry-container** stores items, so this has to be consistent

Comment (wuarmin) [reply lib/dry/system/plugins/dependency_graph.rb:38]: Ok, that makes things clear to me. I'll fix this today.

**Review by solnic — APPROVED** [review]

**Comment by solnic** [comment]
Thanks! This should go out soon as 0.19.1

**Comment by wuarmin** [comment]
Thanks for the merge!

P.S.: If you're interested, I could fix the dry-system-examples (https://github.com/dry-rb/dry-system/tree/master/examples). They are outdated. What do you think?

**Comment by solnic** [comment]
@wuarmin yes that would be very helpful!

**Pull request merged**